force-app/main/default/classes/SOQL.cls

```diff
@@ -286,22 +465,29 @@ public with sharing class SOQL {
      * @param A set of Lead Ids.
      * @return A list of Accounts with matching email domains.
      */
-    public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
+    public static List<Account> findContactsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
+        List<Lead> leadsWithDomain = [SELECT Id, Email FROM Lead WHERE Id IN :leadIds];
 
         // Create a set of email domains from the Leads
+        Set<String> domainsFromLeads = new Set<String>();
 
         // Loop through each Lead
-
+        for(Lead currentLead : leadsWithDomain) {
             // Get the email from the Lead
-
+            String currentEmail = currentLead.Email;
             // Check if the email is not null
-
+            if (currentEmail != null) {
                 // Add the email domain to the set
-
-        
+                currentEmail = currentEmail.split('@').get(1);
+                String domainForAcct = 'www.' + currentEmail;
```

Comment:
You can add the 'www.' in the line above to avoid using another variable:
`currentEmail = 'www.' + currentEmail.split('@').get(1);
`